util/async.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+// TODO(ry) It'd be better to make Deferred a class that inherits from
+// Promise, rather than an interface. This is possible in ES2016, however
```

Comment:
Also, this kicks v8 in a mode where every promise is very slow. Don't do it.

Reply:
I'm just moving this code to a centralized location. I'll leave it to future work to optimize it.